### DISCUSSION TITLE
How to SetResult() an array of struct?

### DISCUSSION THREAD

**Discussion by wsw70** [go-resty/resty#534] [Q&A] [answered]
I am desperately trying to automatically unmarshall data of the form
```
[
  {
    "firstname": "john",
    "lastname": "brown"
  },
(...)
]
```

My code is basically

```
type SMMFormatForUser []struct {
	Firstname string
	Lastname  string
}

func updateXUsers() {
	var allUsers SMMFormatForUser
	r, err := resty.New().R().
		SetResult(allUsers).
		Post("https://example.com")
	log.Debug().Msgf("%v", r.Size())
}
```

The response is OK as I get an expected size (`r.Size()`) but `allUsers` is `nil`.

I tried all kinds of combinations with pointers, trying a plain `struct` and unmarshalling into  `[]*SMMFormatForUser`etc. - the result is always the same.

What is the proper incantation to unmarshal this data?

**Comment by moorereason** [comment]
Quick, untested response:

This trips up most people new to Go unmarshalling.  The HTTP response contains an array, but the response itself is not an array.  Try putting the `SMMFormatForUser` inside another struct.

**Reply by wsw70** [reply]
Thank you. I am not sure I understand. The JSON data I get is an array of objects - how would that fit with that structure?

I now tried 

```
type SMMFormatForUser []struct {
	Firstname string
	Lastname  string
}

type s2 struct {
	Hello SMMFormatForUser
}

func updateXUsers() {
	var allUsers s2
	r, err := resty.New().R().
		SetResult(allUsers).
		Post("https://example.com")
	log.Debug().Msgf("%v", r.Size())
}
```

But the result is the same (`allUsers` is `nil`)

**Comment by stephanwesten** [comment, marked as answer]
For me there was no need to introduce a separate struct

```
rows := []Row{}
....
SetResult(&rows).
...
```
works. I did have to set the return type to json:

`SetHeader("Accept", "application/json").`
